### PR TITLE
New abbreviations.

### PR DESCRIPTION
**Pull request by Yukiiro-ya**
New abbreviations. Very comfortable for players in  Everlasting Summer.

### CODE REVIEW

**Comment by LuqueDaniel** [comment]
Thanks for the PR!

Does the dspr transition exist?

**Comment by Yukiiro-ya** [comment]
In the game Everlasting Summer, yes. And this custom code is very useful there. 
https://store.steampowered.com/app/331470/Everlasting_Summer/
semen_dv_dance

**Comment by Yukiiro-ya** [comment]
I would like to add more new abbreviations. If you do not mind?

**Comment by LuqueDaniel** [comment]
Of course yes. The idea is to improve the extension.

Regarding the snippet `dspr` it doesn't have a function by itself if it is not declared before.

**Comment by Yukiiro-ya** [comment]
The dspr fragment is convenient for players of the specified visual story. She has many players who write custom modifications for her. And those abbreviations that I would like to add are convenient for these modifications.

**Comment by LuqueDaniel** [comment]
Ok, but it would be nice to add a clarification in the description of those snippets.

**Comment by Yukiiro-ya** [comment]
Add a clarification? And where to add it? Yes, and please check whether I added correctly, (window show - window hide) should be so.
snippets

**Comment by LuqueDaniel** [comment]
What do you want the snippet to do?

Like this?

imagen

```json
"body": "window ${1|show,hide|} ${2:placeholder}",
```

**Comment by Yukiiro-ya** [comment]
----Window show.
----'Text'.
----Window hide.

As in the picture.
winow s-h

**Comment by LuqueDaniel** [comment]
```json
"body": ["window show",
              "'$1'",
	      "window hide"],
```

**Comment by Yukiiro-ya** [comment]
Yes, that's right.

File: snippets/snippets.json [hunk 1]
```diff
@@ -149,6 +154,11 @@
         "body": "while ${1:condition}:\n\t${2:#block of code to run}",
         "description": "While (Loop)"
       },
+      "Window": {
+        "prefix": "window",
+        "body": "window show\n${1:""}:\n${2:window hide}",
```
Comment (LuqueDaniel) [inline snippets/snippets.json]: The snippet doesn't work.

File: snippets/snippets.json [hunk 2]
```diff
@@ -99,6 +99,11 @@
         "body": "with dissolve",
         "description": "Dissolve"
       },
+      "Dspr": {
+        "prefix": "wDspr",
+        "body": "with dspr",
+        "description": "Dspr"
```
Comment (LuqueDaniel) [inline snippets/snippets.json]: It would be good to improve the description.

Comment (Yukiiro-ya) [reply snippets/snippets.json]: Ok I'll add

File: snippets/snippets.json (hunk 1 above)
Comment (Yukiiro-ya) [reply snippets/snippets.json]: Why does not it work?

Comment (LuqueDaniel) [reply snippets/snippets.json]: I don't know, but when I use it :

imagen

Comment (Yukiiro-ya) [reply snippets/snippets.json]: Hmm weird

**Comment by Yukiiro-ya** [comment]
Безымянный

File: snippets/snippets.json (hunk 2 above)
Comment (Yukiiro-ya) [reply snippets/snippets.json]: added description

File: snippets/snippets.json (hunk 1 above)
Comment (Yukiiro-ya) [reply snippets/snippets.json]: the problem is still there

File: snippets/snippets.json [hunk 3]
```diff
@@ -109,6 +109,11 @@
         "body": "with pixellate",
         "description": "Pixellate"
       },
+      "Music": {
+        "prefix": "playMusic",
+        "body": "play music music_list['#selected composition'] fadein 2 #fadein, if need a delay",
```
Comment (LuqueDaniel) [inline snippets/snippets.json]: Please use tabstops, placeholders and choices, when necessary

[Visual Studo Code Snippets Syntax](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax)

File: snippets/snippets.json (hunk 1 above)
Comment (LuqueDaniel) [reply snippets/snippets.json]: I see it, I'll wait for you to fix it

File: snippets/snippets.json [hunk 4]
```diff
@@ -111,7 +111,7 @@
       },
       "Music": {
         "prefix": "playMusic",
-        "body": "play music music_list['#selected composition'] fadein 2 #fadein, if need a delay",
+        "body": "play music music_list['$1 #selected composition'] fadein 2 $2 #fadein, if need a delay",
```
Comment (LuqueDaniel) [inline snippets/snippets.json]: Broken. The comment "#fadein, if need a delay" isn't necessary. Incorrect syntax for tabstops and placeholders, should be like: `${1:placeholder}`.

File: snippets/snippets.json [hunk 5]
```diff
@@ -161,7 +161,7 @@
       },
       "Window": {
         "prefix": "window",
-        "body": "window show\n${1:""}:\n${2:window hide}",
+        "body": "window show\n${1:"#text - author, character or thoughts"}:\n${2:window hide}",
```
Comment (LuqueDaniel) [inline snippets/snippets.json]: You forgot to escape the double quotes. Syntax error, remove the colon `:`.

Comment (Yukiiro-ya) [reply snippets/snippets.json]: Added slant slash this screen for quotes.

Comment (Yukiiro-ya) [reply snippets/snippets.json]: I want to add characters from Everlasting Summer if you do not mind.

**Comment by LuqueDaniel** [comment]
I'll close this PR because I don't think it goes in one direction

**Pull request closed without merging**